azurerm/internal/services/web/app_service_certificate_resource.go

```diff
@@ -74,12 +74,17 @@ func resourceAppServiceCertificate() *pluginsdk.Resource {
 				ConflictsWith: []string{"pfx_blob", "password"},
 			},
 
-			"hosting_environment_profile_id": {
+			"app_service_plan_id": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
 				ForceNew: true,
 			},
 
+			"hosting_environment_profile_id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
```

Comment:
Unfortunately this would be a breaking change, as users that have this in config (despite not working as intended) would hit a configuration error, we can avoid this by:
```suggestion
			"hosting_environment_profile_id": { // TODO - Remove in 3.0
				Type:       pluginsdk.TypeString,
				Optional:   true,
				Computed:   true,
				Deprecated: "This property has been deprecated and replaced with `app_service_plan_id`",
			},
			
```